packages/xml-for-vscode/test/suite/extension.test.js

```diff
@@ -0,0 +1,129 @@
+    const content = `
+      <note!>
+        <to>Tove</to>
+        <from>Jani</from>
+        <heading>Reminder</heading>
+        <body>Don't forget me this weekend!</body>
+      </note>`;
+    await setContent(content);
+    await testDiagnostics(docUri, [
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message:
+          "unexpected character: ->!<- at offset: 12, skipped 1 characters.",
+        range: toRange(1, 11, 1, 12),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  it("Will show parsing errors in problems view", async function() {
+    const content = `
+      <note>
+        <to a=>Tove</to>
+        <from>Jani</from>
+        <heading>Reminder</heading>
+        <body>Don't forget me this weekend!</body>
+      </note`;
+    await setContent(content);
+    await testDiagnostics(docUri, [
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(2, 14, 2, 14),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> CLOSE <-- but found --> '' <--",
+        range: toRange(6, null, 0, 0),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  it("Will show multiple errors in problems view", async function() {
+    const content = `
+      <note>
+        <to a=>Tove</to>
+        <from a=>Jani</from>
+        <heading a=>Reminder</heading>
+        <body>Don't forget me this weekend!</body>
+      </note>`;
+    await setContent(content);
+    await testDiagnostics(docUri, [
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(2, 14, 2, 14),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(3, 16, 3, 16),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(4, 19, 4, 19),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  function toRange(sLine, sChar, eLine, eChar) {
+    const start = new vscode.Position(sLine, sChar);
+    const end = new vscode.Position(eLine, eChar);
+    return new vscode.Range(start, end);
+  }
+
+  async function setContent(content) {
+    await fs.writeFile(docPath, content);
+    await sleep(1000);
+  }
+
+  async function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  async function testDiagnostics(docUri, expectedDiagnostics) {
+    const actualDiagnostics = vscode.languages.getDiagnostics(docUri);
+    expect(expectedDiagnostics).to.deep.equal(actualDiagnostics);
```

Comment:
use: https://www.chaijs.com/plugins/deep-equal-in-any-order/

It will avoid tests breaking due to possible changes in language-server implementation